ui/src/dashboards/components/CellEditorOverlay.js

```diff
@@ -63,6 +73,111 @@ class CellEditorOverlay extends Component {
     }
   }
 
+  handleAddThreshold = () => {
+    const {colors} = this.state
+
+    if (colors.length <= MAX_THRESHOLDS) {
+      const randomColor = _.random(0, GAUGE_COLORS.length)
+
+      const maxValue = Number(
+        colors.find(color => color.type === COLOR_TYPE_MAX).value
+      )
+      const minValue = Number(
+        colors.find(color => color.type === COLOR_TYPE_MIN).value
+      )
+
+      const colorsValues = _.mapValues(colors, 'value')
+      let randomValue
+
+      do {
+        randomValue = String(_.round(_.random(minValue, maxValue, true), 2))
+      } while (_.includes(colorsValues, randomValue))
+
+      const newThreshold = {
+        type: COLOR_TYPE_THRESHOLD,
+        id: uuid.v4(),
+        value: randomValue,
+        hex: GAUGE_COLORS[randomColor].hex,
+        name: GAUGE_COLORS[randomColor].name,
+      }
+
+      this.setState({colors: [...colors, newThreshold]})
+    }
+  }
+
+  handleDeleteThreshold = threshold => () => {
+    const {colors} = this.state
+
+    const newColors = colors.filter(color => color.id !== threshold.id)
+
+    this.setState({colors: newColors})
+  }
+
+  handleChooseColor = threshold => chosenColor => {
+    const {colors} = this.state
+
+    const newColors = colors.map(
+      color =>
+        color.id === threshold.id
+          ? {...color, hex: chosenColor.hex, name: chosenColor.name}
+          : color
+    )
+
+    this.setState({colors: newColors})
+  }
+
+  handleUpdateColorValue = (threshold, newValue) => {
+    const {colors} = this.state
+    const newColors = colors.map(
+      color => (color.id === threshold.id ? {...color, value: newValue} : color)
+    )
+    this.setState({colors: newColors})
+  }
+
+  handleValidateColorValue = (threshold, e) => {
+    const {colors} = this.state
+    const sortedColors = _.sortBy(colors, color => Number(color.value))
+    const targetValueNumber = Number(e.target.value)
+
+    const maxValue = Number(
+      colors.find(color => color.type === COLOR_TYPE_MAX).value
+    )
+    const minValue = Number(
+      colors.find(color => color.type === COLOR_TYPE_MIN).value
+    )
```

Comment:
Perhaps we can talk about our coercion methods, this more explicit pattern is actually kinda interesting.

Reply:
@121watts what do you think? `Number()` constructor or `+`-prefix type coercion?

Reply:
I'm a fan of `+`, mostly because its shorter.  I'm fine with either.